x-pack/solutions/observability/plugins/observability/server/services/related_dashboards_client.ts

```diff
@@ -4,383 +4,74 @@
-
-  private getPanelIndices(panel: DashboardPanel): Set<string> {
-    const indices = isSuggestedDashboardsValidPanelType(panel.type)
-      ? this.getPanelIndicesMap[panel.type](panel)
-      : undefined;
-    return indices ?? new Set<string>();
-  }
-
-  private getPanelFields(panel: DashboardPanel): Set<string> {
-    const fields = isSuggestedDashboardsValidPanelType(panel.type)
-      ? this.getPanelFieldsMap[panel.type](panel)
-      : undefined;
-    return fields ?? new Set<string>();
-  }
-
-  private isLensVizAttributes(attributes: unknown): attributes is LensAttributes {
-    if (!attributes) {
-      return false;
-    }
-    return (
-      Boolean(attributes) &&
-      typeof attributes === 'object' &&
-      'type' in attributes &&
-      attributes.type === 'lens'
-    );
-  }
-
-  private getRuleQueryIndex(): string | null {
-    const alert = this.checkAlert();
-    const index = alert.getRuleQueryIndex();
-    return index;
-  }
-
-  private async getLinkedDashboards(): Promise<LinkedDashboard[]> {
-    const alert = this.checkAlert();
     const ruleId = alert.getRuleId();
     if (!ruleId) {
       throw new Error(
         `Alert with id ${this.alertId} does not have a rule ID. Could not fetch linked dashboards.`
       );
     }
-    const rule = await this.alertsClient.getRuleById(ruleId);
-    if (!rule) {
-      throw new Error(
-        `Rule with id ${ruleId} not found. Could not fetch linked dashboards for alert with id ${this.alertId}.`
-      );
-    }
-    const linkedDashboardsArtifacts = rule.artifacts?.dashboards || [];
-    const linkedDashboards = await this.getLinkedDashboardsByIds(
-      linkedDashboardsArtifacts.map((d) => d.id)
-    );
-    return linkedDashboards;
-  }
-
-  private async getLinkedDashboardsByIds(ids: string[]): Promise<LinkedDashboard[]> {
-    const linkedDashboardsResponse = await Promise.all(
-      ids.map((id) => this.getLinkedDashboardById(id))
-    );
-    return linkedDashboardsResponse.filter((dashboard): dashboard is LinkedDashboard =>
-      Boolean(dashboard)
-    );
-  }
-
-  private async getLinkedDashboardById(id: string): Promise<LinkedDashboard | null> {
-    try {
-      const dashboardResponse = await this.dashboardClient.get(id);
-      return {
-        id: dashboardResponse.result.item.id,
-        title: dashboardResponse.result.item.attributes.title,
-        matchedBy: { linked: true },
-        description: dashboardResponse.result.item.attributes.description,
-        tags: dashboardResponse.result.item.attributes.tags,
-      };
-    } catch (error) {
-      if (error.output.statusCode === 404) {
-        this.logger.warn(`Linked dashboard with id ${id} not found. Skipping.`);
-        return null;
-      }
-      throw new Error(`Error fetching dashboard with id ${id}: ${error.message || error}`);
-    }
-  }
+    const linkedDashboardIds = (
+      (await await this.alertsClient.getRuleById(ruleId))?.artifacts?.dashboards || []
+    ).map((d) => d.id);
```

Comment:
```suggestion
    const linkedDashboardIds = (
      (await this.alertsClient.getRuleById(ruleId))?.artifacts?.dashboards || []
    ).map((d) => d.id);
```

I think we do not want this `await await`